### PR TITLE
Server - App - Implement scheduler module

### LINKED ISSUE #64: Server - App - Create scheduler module
- [ ] Create a scheduler structure that will bind an `uuid` to an amount of `ticks` that the binded `uuid` will have to wait.
- [ ] Create helper functions to interact with the scheduler
  - [ ] `scheduler_schedule_event(uuid, ticks)` that returns a `bool` wether or not the action was scheduled
  - [ ] `scheduler_update(entity ** map)` that update all the entity on the map that where bound to an event in the scheduler
  - [ ] `scheduler_has_event(uuid)` that returns a `bool` wether or not the given `uuid` has a corresponding event bound to itself

### PR DESCRIPTION
**Pull request by Cavonstavant**
- Implements a scheduler data structure that binds an `uuid_t` to an amount of `ticks` to wait
- Create, Schedule, Check, Update scheduled events

Closes #64

### CODE REVIEW

File: src/zappy_server_src/core/include/scheduler/scheduler.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAAAAAAAAAAAAppy
+** File description:
+** scheduler
+*/
+
```
Comment (osvegn) [inline src/zappy_server_src/core/include/scheduler/scheduler.h:7]: missing \file

File: src/zappy_server_src/core/include/scheduler/scheduler.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+/// \brief represent a scheduler event that will be stored inside the scheduler
+typedef struct scheduler_event_s {
+    /// \brief The entity uuid bound to the event
+    uuid_t id;
+
+    /// \brief The amount of time to wait before the event is over
+    int ticks;
+
+    TAILQ_ENTRY(scheduler_event_s) events;
```
Comment (osvegn) [inline src/zappy_server_src/core/include/scheduler/scheduler.h:23]: missing documentation

File: src/zappy_server_src/core/include/scheduler/scheduler.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+} scheduler_event_t;
+
+TAILQ_ENTRY(scheduler_event_s) scheduler_event_head;
```
Comment (osvegn) [inline src/zappy_server_src/core/include/scheduler/scheduler.h:26]: missing documentation (Check all the .h file, I won't wrote "missing documentation" each time)

File: src/zappy_server_src/core/scheduler.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAAAAAAAAAAAAppy
+** File description:
+** scheduler
+*/
+
```
Comment (osvegn) [inline src/zappy_server_src/core/scheduler.c:7]: missing \file

File: src/zappy_server_src/core/scheduler.c [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    TAILQ_FOREACH(tmp, &self->events, events)
+        if (uuid_compare(tmp->id, uuid) == 0)
+            return true;
+    return false;
+}
+
+bool scheduler_schedule_event(scheduler_t *self, uuid_t uuid, int ticks)
+{
+    scheduler_event_t *new_event = NULL;
+    if (!self)
```
Comment (osvegn) [inline src/zappy_server_src/core/scheduler.c:38]: missing empty line

**Review by osvegn — CHANGES_REQUESTED** [review]
Missing documentation and an L6 norme error

File: src/zappy_server_src/core/scheduler.c [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+void scheduler_update(scheduler_t *self)
+{
+    scheduler_event_t *tmp = NULL;
+    time_t now = time(NULL);
+
+    if (!self)
+        return;
+    TAILQ_FOREACH(tmp, &self->events, events) {
+        tmp->ticks -= (now - self->clock) * self->freq;
```
Comment (LucasTesnier) [inline src/zappy_server_src/core/scheduler.c:58]: Are you sure about this ?

**Review by LucasTesnier — APPROVED** [review]
Just a question, otherwise it's okay for me

File: src/zappy_server_src/core/scheduler.c (hunk 6 above)
Comment (Cavonstavant) [reply src/zappy_server_src/core/scheduler.c:58]: I'll get back to you on this topic ASAP

**Review by 31Nathan — APPROVED** [review]
Change all the things that have been asked by osvegn, otherwise it's good

File: src/zappy_server_src/core/CMakeLists.txt [hunk 7]
```diff
@@ -26,4 +28,6 @@ target_include_directories(${TARGET}
 
 target_link_libraries(${TARGET}
     PRIVATE net_utils
+    PRIVATE uuid
+    PRIVATE m
```
Comment (osvegn) [inline src/zappy_server_src/core/CMakeLists.txt:32]: What is the "m" library ?

**Review by osvegn — CHANGES_REQUESTED** [review]
Missing one variable documentation, and need to resolve a conflict, otherwise all look good to me

**Review by osvegn — APPROVED** [review]
Nice job, look good to me

**Pull request merged**